### PR TITLE
chore: Added confirmation dialog for config import.

### PR DESCRIPTION
**Pull request by ErikSin**
Adds a confirmation dialog when a config has been successfully imported.

Kapture 2021-11-29 at 18 11 23

### CODE REVIEW

File: src/main/menu.js [hunk 1]
```diff
@@ -80,16 +81,27 @@ function menuTemplate (ipc) {
...
-                if (err) logger.error(err)
-                logger.debug('[SYSTEM] Forcing window refresh')
+              ipc.send('reload-config', async (err) => 
+              {
+                if (err) 
+                {
+                  logger.error(err)
+                  logger.debug('[SYSTEM] Forcing window refresh')
+                  focusedWindow.webContents.send('force-refresh-window')
+                  logger
```
Comment (achou11) [inline src/main/menu.js:94]: need to remove this line since it's causing lint errors

File: src/main/menu.js [hunk 2]
```diff
@@ -80,16 +81,27 @@ function menuTemplate (ipc) {
...
+                  return
+                }
+                await dialog.showMessageBox({message:t('menu-config-complete'), buttons:[t("button-submit")]})
```
Comment (achou11) [inline src/main/menu.js:97]: think we'd ideally show the name of the config that's been imported (similar to [mobile](https://github.com/digidem/mapeo-mobile/pull/864)). Would require a little bit of work in [`user-config.js`](https://github.com/digidem/mapeo-desktop/blob/660754ea6483064f6f909390d010245a18461bb1/src/main/user-config.js#L7) to pass the settings metadata

**Comment by achou11** [comment]
Can you make sure your editor has the formatter set up correctly? Looks like the changed JS file doesn't conform to Standard's formatting (it should also run automatically as a pre-commit step) 🤔

File: messages/main/en.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
   "menu-file": "File",
   "menu-change-language": "Change language...",
   "menu-change-language-title": "Enter the language (e.g., 'en' or 'es')",
+  "menu-config-complete": "New config has been loaded",
```
Comment (achou11) [inline messages/main/en.json:12]: Based on https://github.com/digidem/mapeo-desktop/issues/618, message should be `Successfully imported config:`

**Comment by ErikSin** [comment]
> Can you make sure your editor has the formatter set up correctly? Looks like the changed JS file doesn't conform to Standard's formatting (it should also run automatically as a pre-commit step) 🤔
> 
> (Some of it was also probably pre-existing)

Oh weird, Yeah it looks like it is not formatting in the pre-commit step. I turned off my formatter due to our automation. Husky is suppose to be doing the pre-commit step, correct?

File: src/main/menu.js [hunk 4]
```diff
@@ -80,16 +81,27 @@ function menuTemplate (ipc) {
...
                 focusedWindow.webContents.send('force-refresh-window')
+
               })
             })
 
-            function onerror (err) {
+            async function onerror (err)
```
Comment (achou11) [inline src/main/menu.js:103]: don't think you need `async` here since the function doesn't need to return a promise

**Comment by achou11** [comment]
> Oh weird, Yeah it looks like it is not formatting in the pre-commit step. I turned off my formatter due to our automation. Husky is suppose to be doing the pre-commit step, correct?

yeah Husky should be installed and working if you did an `npm i`

File: src/main/menu.js (hunk 4 above)
Comment (ErikSin) [reply src/main/menu.js:103]: good catch. I the other dialog are async, so I awaited it, and then realized it was not returning a promise. took out the await, not the async

**Comment by ErikSin** [comment]
It looks the precommit step is working now (I ran `npm i`)

File: messages/main/en.json [hunk 5]
```diff
@@ -9,6 +9,7 @@
   "menu-file": "File",
   "menu-change-language": "Change language...",
   "menu-change-language-title": "Enter the language (e.g., 'en' or 'es')",
+  "menu-config-complete": "Successfully Imported Config: ",
```
Comment (achou11) [inline messages/main/en.json:12]: Should be sentence-cased:

```suggestion
  "menu-config-complete": "Successfully imported config: ",
```

File: src/main/menu.js (hunk 1 above)
Comment (achou11) [reply src/main/menu.js:85]: still need to remove this line

Comment (ErikSin) [reply src/main/menu.js:85]: What is the linting error here. I didn't add this, I just changed the formatting, and thats why it looks I made changes

Comment (ErikSin) [reply src/main/menu.js:85]: OHH nvm I see it

Comment (achou11) [reply src/main/menu.js:85]: line 85 has a standalone line that you added of just `logger`

**Review by achou11 — APPROVED** [review]
Mind updating the screenshot to reflect latest changes?

**Pull request merged**